src/prefect/engine.py

```diff
@@ -1906,6 +1921,16 @@ def should_log_prints(flow_or_task: Union[Flow, Task]) -> bool:
             f"Engine execution of flow run '{flow_run_id}' is paused: {exc}"
         )
         exit(0)
+    except TerminationSignal as exc:
+        engine_logger.info(
+            f"Engine execution of flow run '{flow_run_id}' cancelled by external signal: {exc}"
+        )
+
+        # Termination signals are swapped out during a flow run to perform a
+        # graceful shutdown and raise this exception. This `os.kill` call
+        # ensures that the previous handler, likely the Python default, gets
+        # called as well.
+        os.kill(os.getpid(), exc.signal)
```

Comment:
This seems like it should be done by `report_flow_run_crashes` — otherwise the original handler is only called for deployed flow runs.